pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -43,14 +51,18 @@ type fatalError string
 type transientError string
 
 func (e fatalError) Error() string {
-	return fmt.Sprintf("fatal error handling DeploymentConfig: %s", string(e))
+	return fmt.Sprintf("fatal error handling deploymentConfig: %s", string(e))
 }
 func (e transientError) Error() string {
-	return "transient error handling DeploymentConfig: " + string(e)
+	return "transient error handling deploymentConfig: " + string(e)
 }
 
 // Handle processes config and creates a new deployment if necessary.
 func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig) error {
+	// Inspect a deployment configuration every time the controller reconciles it
+	if details := c.failureInspection(config); len(details) > 0 {
+		c.detailsUpdate(config, details)
```

Comment:
Closer... there are a few mods to fit the general reconcilation model of other controllers. Something like:

```go
details := c.findDetails(config, client)
c.updateDetails(config, details)
```

The `findDetails` function could immediately check to see if there's a deployment for `config`, and if so, return an empty string (to cause details to be cleared). It could then do all its other checks. Then `findDetails` needs to only be called once, from here (the one on L94 would not be necessary).

The `updateDetails` function would update config if `details != config.details.Message`. That handles setting new messages and clearing outdated ones, and handles no-op'ing if there's no difference.

You should also ditch all those Namespacers and just use a single `client.Interface`. The test support works just as well with those but with a lot less code.